flink-doris-connector/src/main/java/org/apache/doris/flink/sink/writer/serializer/JsonDebeziumSchemaSerializer.java

```diff
@@ -156,13 +161,25 @@ public DorisRecord serialize(String record) throws IOException {
             return null;
         }
 
-        if (firstLoad) {
-            schemaChange.init(recordRoot);
-            firstLoad = false;
+        Map<String, String> tableMapping = schemaChange.getTableMapping();
+        if (initSchemaChange(firstLoad, tableMapping)) {
+            schemaChange.init(recordRoot, initTableSet);
+            this.firstLoad = false;
         }
+
         return dataChange.serialize(record, recordRoot, op);
     }
 
+    private boolean initSchemaChange(boolean firstLoad, Map<String, String> tableMapping) {
+        if (firstLoad) {
+            return true;
+        }
+        tableMappingSet.clear();
+        tableMappingSet.addAll(tableMapping.keySet());
+        tableMappingSet.removeAll(initTableSet);
```

Comment:
Is it sufficient to determine if the table exists in tableMapping?

Reply:
Sure, tableMapping is built when the program starts. When a new table is added, the latest tableMapping will be maintained in `JsonDebeziumSchemaChange`